apm-agent-core/src/main/java/co/elastic/apm/agent/universalprofiling/UniversalProfilingIntegration.java

```diff
@@ -114,4 +167,65 @@ public void correlateAndReport(Transaction endedTransaction) {
     public void drop(Transaction endedTransaction) {
         //TODO: remove dropped transactions from correlation storage without reporting
     }
+
+
+    private String openProfilerSocket(String socketDir) {
+        Path dir = Paths.get(socketDir);
+        if (!Files.exists(dir) && !dir.toFile().mkdirs()) {
+            throw new IllegalArgumentException("Could not create directory '" + socketDir + "'");
+        }
+        Path socketFile;
+        do {
+            socketFile = dir.resolve(randomSocketFileName());
+        } while (Files.exists(socketFile));
+
+        String absolutePath = socketFile.toAbsolutePath().toString();
+        log.debug("Opening profiler correlation socket {}", absolutePath);
+        UniversalProfilingCorrelation.startProfilerReturnChannel(absolutePath);
+        return absolutePath;
+    }
+
+    private String randomSocketFileName() {
+        StringBuilder name = new StringBuilder("essock");
+        String allowedChars = "abcdefghijklmonpqrstuvwxzyABCDEFGHIJKLMONPQRSTUVWXYZ0123456789";
+        Random rnd = new Random();
+        for (int i = 0; i < 8; i++) {
+            int idx = rnd.nextInt(allowedChars.length());
+            name.append(allowedChars.charAt(idx));
+        }
+        return name.toString();
+    }
+
+    private void consumeProfilerMessages() {
+        try {
+            while (true) {
```

Comment:
should this be while(isActive)

Reply:
No, the intention of the loop is to drain all available messages. The loop can be aborted by either of the following two conditions:
 * No more messages are available (the returned message is `null`)
 * There is some problem (except for a single message being non decodeable, represented by the `DecodeException`). The loop will be aborted by an exception.